src/metadata/modules.cpp

```diff
@@ -484,11 +484,29 @@ static HRESULT LoadSymbols(IMetaDataImport *pMD, ICorDebugModule *pModule, VOID
     IfFailRet(pModule->GetBaseAddress(&peAddress));
     IfFailRet(pModule->GetSize(&peSize));
 
+    std::vector<unsigned char> peBuf;
+    ULONG64 peBufAddress = 0;
+    if (isInMemory)
+    {
+        ICorDebugProcess* process = 0;
+        IfFailRet(pModule->GetProcess(&process));
+
+        if (peAddress != 0 && peSize != 0)
+        {
+            peBuf.resize(peSize);
+            peBufAddress = (ULONG64)&peBuf[0];
+            SIZE_T read = 0;
+            IfFailRet(process->ReadMemory(peAddress, peSize, &peBuf[0], &read));
```

Comment:
If I am correct, this mean it copy module from debuggee process into debugger process. Is it copy all modules or in-memory modules only?

Reply:
I mean, this looks like you copy all, and this is not really good, since could take a lot of time for big programs on weak devices (like TV or RPI). Plus, if case `ReadMemory` fail with copy for normal module ->`return E_FAIL;`... but why, this is not issue for normal (not in-memory) module that load data from DLL file in this case.

Reply:
I tested it just now and can verify on my end that this section only copies the in memory module. I understand the performance concern but not entirely sure if there is a better way to do it. Least that can be done is limit it to only the in-memory modules.

Reply:
As I see in MS docs, 
https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugmodule-getbaseaddress-method
```
If the module is a native image (that is, if the module was produced by the native image generator, NGen.exe), its base address will be zero.
```
https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugmodule-getsize-method
```
If the module was produced from the native image generator (NGen.exe), the size of the module will be zero.
```
Did you tested with normal modules, not native images?
As I know, this calls return proper data (not zero) and we also use them in other debugger code (stepper/breakpoint/etc) for Debug built modules (that is not native image for sure).

Not sure how in-memory modules should be detected, but `if (peAddress != 0 && peSize != 0)` not work for this for sure.

Reply:
Ohh... miss `IsInMemory` call.